### PR TITLE
fix(metrics_extraction): `epm` and `eps` to use standard metrics when not applicable

### PR DESCRIPTION
**Pull request by armenzg**
The task would call `should_use_on_demand` with `epm()` while the endpoint would call with `epm(900)`. In either case, we should be using standard metrics since the query is empty.

### CODE REVIEW

File: src/sentry/snuba/metrics/extraction.py [hunk 1]
```diff
@@ -689,6 +689,9 @@ def _get_args_support(fields: Sequence[str], used_in_function: str | None = None
         # apdex can have two variations, either apdex() or apdex(value).
         return SupportedBy(on_demand_metrics=True, standard_metrics=False)
 
+    if used_in_function in ["epm", "eps"]:
+        return SupportedBy.both()
```
Comment (armenzg) [inline src/sentry/snuba/metrics/extraction.py:693]: This is the actual fix since `epm` and `eps` are called with the field `900` since the endpoint adds it.

File: src/sentry/snuba/metrics/extraction.py [hunk 2]
```diff
@@ -1063,12 +1066,10 @@ def user_misery_tag_spec(project: Project, arguments: Sequence[str] | None) -> l
 
 
 # This is used to map a metric to a function which generates a specification
-_DERIVED_METRICS: dict[MetricOperationType, TagsSpecsGenerator | None] = {
+_DERIVED_METRICS: dict[MetricOperationType, TagsSpecsGenerator] = {
     "on_demand_failure_count": failure_tag_spec,
     "on_demand_failure_rate": failure_tag_spec,
     "on_demand_apdex": apdex_tag_spec,
-    "on_demand_epm": None,
-    "on_demand_eps": None,
```
Comment (armenzg) [inline src/sentry/snuba/metrics/extraction.py:1071]: This was an unnecessary change in my original PR.

File: tests/sentry/relay/config/test_metric_extraction.py [hunk 3]
```diff
@@ -1564,16 +1564,27 @@ def test_get_metric_extraction_config_with_unicode_character(default_project: Pr
 
 
 @django_db_all
-@pytest.mark.parametrize("metric", [("epm()"), ("eps()")])
-def test_get_metric_extraction_config_with_no_tag_spec(
```
Comment (armenzg) [inline tests/sentry/relay/config/test_metric_extraction.py:1568]: The name of the function did not make sense.

File: tests/sentry/relay/config/test_metric_extraction.py [hunk 4]
```diff
@@ -1564,16 +1564,27 @@ def test_get_metric_extraction_config_with_unicode_character(default_project: Pr
...
-    default_project: Project, metric: str
+@pytest.mark.parametrize(
+    "metric,query,query_hashes",
+    [
+        ("epm()", "transaction.duration:>=1000", ["8f8293cf", "5200e087"]),
+        ("eps()", "transaction.duration:>=1000", ["9ffdd8ac", "162178e9"]),
+        ("epm()", "", []),
```
Comment (armenzg) [inline tests/sentry/relay/config/test_metric_extraction.py:1572]: Introducing an extra test case for when we should be defaulting to standard metrics.

File: tests/sentry/snuba/metrics/test_extraction.py [hunk 5]
```diff
@@ -141,24 +148,18 @@ def test_creates_on_demand_spec(self, aggregate, query) -> None:
         "aggregate, query",
         [
             ("count()", "release:a"),  # supported by standard metrics
-            (
```
Comment (armenzg) [inline tests/sentry/snuba/metrics/test_extraction.py:144]: Just some flattening of this test inputs.

File: tests/snuba/api/endpoints/test_organization_events_stats_mep.py [hunk 6]
```diff
@@ -1410,6 +1410,30 @@ def test_is_metrics_extracted_data_is_included(self):
...
+            "yAxis": ["epm()"],
+        }
+        response = self.do_request(params)
+
+        assert response.status_code == 200, response.content
+        assert response.data["meta"] == {
+            "fields": {"time": "date", "epm_900": "rate"},
+            "units": {"time": None, "epm_900": None},
+            "isMetricsData": True,
+            "isMetricsExtractedData": False,
```
Comment (armenzg) [inline tests/snuba/api/endpoints/test_organization_events_stats_mep.py:1431]: Without the change in `src/` this would be `True`.

File: tests/snuba/api/endpoints/test_organization_events_stats_mep.py [hunk 7]
```diff
@@ -1410,6 +1410,30 @@ def test_is_metrics_extracted_data_is_included(self):
...
+            "query": "",
+            "statsPeriod": "1h",
+            "useOnDemandMetrics": "true",
+            "yAxis": ["epm()"],
+        }
+        response = self.do_request(params)
+
+        assert response.status_code == 200, response.content
+        assert response.data["meta"] == {
+            "fields": {"time": "date", "epm_900": "rate"},
```
Comment (armenzg) [inline tests/snuba/api/endpoints/test_organization_events_stats_mep.py:1428]: Notice how the endpoint has added `_900`.

**Review by k-fish — APPROVED** [review]
Awesome 👍
